### PR TITLE
Fix HTTP 431 issues

### PR DESCRIPTION
**Pull request by tahmed-wish**
Follow up to: https://github.com/ContextLogic/gohive/pull/1

For some reason the data was not visible in Tahoe. Not sure what the issue is but this new implementation seems to work 😅

### CODE REVIEW

File: hive.go [hunk 1]
```diff
@@ -1220,20 +1220,47 @@ func getTotalRows(columns []*hiveserver.TColumn) (int, error) {
...
-	return jar.storage[u.Host]
+	cookiesArray := []*http.Cookie{}
+	for pattern, cookies := range jar.storage {
+		if strings.Contains(u.String(), pattern) {
+			for i := range cookies {
+				cookiesArray = append(cookiesArray, &cookies[i])
+			}
+
+			// Delete from storage
+			delete(jar.storage, pattern)
```
Comment (pchau-wish) [inline hive.go:1247]: hmm can you delete from the map safely while iterating over it?

nvm looked it up, yes you can

File: hive.go [hunk 2]
```diff
@@ -1220,20 +1220,47 @@ func getTotalRows(columns []*hiveserver.TColumn) (int, error) {
 }
 
 type inMemoryCookieJar struct {
-	storage map[string][]*http.Cookie
+	previousStorage map[string]bool
+	storage map[string][]http.Cookie
```
Comment (pchau-wish) [inline hive.go:1224]: hmm cant you just store it as []*http.Cookie so you dont need to convert on line 1243 and 1234?

would also allow you to remove the for loop on 1242 and just do append(cookies...)

**Review by pchau-wish — APPROVED** [review]
just minor nits should be fine like this

File: hive.go (hunk 1 above)
Comment (tahmed-wish) [reply hive.go:1247]: Yup!

File: hive.go (hunk 2 above)
Comment (tahmed-wish) [reply hive.go:1224]: You can but I wanted to follow closer to the initial implementation of the inMemoryCookieJar 😅 

https://github.com/beltran/gohive/blob/master/hive.go#L1251

Comment (pchau-wish) [reply hive.go:1224]: :+1: np should work either way

**Pull request merged**